src/main/java/com/regnosys/testing/transform/TransformTestExtension.java

```diff
@@ -122,9 +132,29 @@ public void beforeAll(ExtensionContext context) {
         this.injector = Guice.createInjector(runtimeModule);
         this.injector.injectMembers(this);
         ClassLoader classLoader = this.getClass().getClassLoader();
-        this.pipelineModel = getPipelineModel(getPipelineModels(configPath, classLoader, JSON_OBJECT_MAPPER), funcType.getName());
-        this.inputObjectMapper = getObjectMapper(pipelineModel.getInputSerialisation()).orElse(JSON_OBJECT_MAPPER);
-        this.outputObjectWriter = getObjectWriter(pipelineModel.getOutputSerialisation()).orElse(JSON_OBJECT_WRITER);
+        List<PipelineModel> pipelineModels = getPipelineModels(configPath, classLoader, JSON_OBJECT_MAPPER);
+        List<PipelineModel> populatedModels = new ArrayList<>();
+        for (PipelineModel model : pipelineModels) {
+            populatedModels.addAll(getPipelineModel(model.getId(), pipelineModels, funcType.getName()));
```

Comment:
I am confused by this. First you are fetching all pipeline models from the class path, and then for each model, you are filtering those pipeline models based on its own id and additionally checking the function name? Is this just filtering out the models based on the function name? If so - this could be done directly.